pest/src/parser.rs

```diff
@@ -8,12 +8,14 @@
 use std::rc::Rc;
 
 use super::error::Error;
-use super::inputs::Input;
+use super::inputs::{Input, StringInput};
 use super::iterators::Pairs;
 use super::RuleType;
 
 /// A `trait` that defines a `Parser`.
 pub trait Parser<R: RuleType> {
     /// Parses `input` starting from `rule`.
     fn parse<I: Input>(rule: R, input: Rc<I>) -> Result<Pairs<R, I>, Error<R, I>>;
+    /// Parses an `input` string starting from `rule`.
+    fn parse_str(rule: R, input: &str) -> Result<Pairs<R, StringInput>, Error<R, StringInput>>;
```

Comment:
So I guess I don't understand what you mean by "I would add the full implementation in Parser instead of keeping it abstract." if it should be part of the trait.